files/usr/lib/jeos-firstboot

```diff
@@ -338,110 +361,9 @@ for d in ${candidates[@]} ; do
-    #
-    get_wlan_devices()
-    {
-        list=()
-        local line
-        while read line; do
-                list+=("${line}" '') 
-        done < <(ls -d /sys/class/net/*/wireless | awk -F'/' '{ print $5 }')
-        [ -n "$list" ]
-    }
-
-    get_wlan_networks()
-    {
-        list=()
-        local line
-        while read line; do
-                list+=("${line}" '') 
-        done < <(ip link set $wlan_device up && iwlist $wlan_device scan|grep ESSID|cut -d':' -f2|cut -d'"' -f2|sort -u)
-        [ -n "${list[*]}" ]
-    }
-
-    wlan_error(){
-        d --title $"Error" --msgbox "$1" 0 0
-        dialog --yesno $"Do you want to retry?" 0 0
-    }
-    while true
-    do
-        if get_wlan_devices && d --menu  $"Select Wireless card to configure" 0 0 $dh_menu "${list[@]}"; then
-            wlan_device="$result"
-        else
-            if wlan_error $"Error listing wlan devices"; then
-                continue
-            else
-                break
-            fi
-        fi
-
-        if get_wlan_networks && d --menu  $"Select Wireless network to connect" 0 0 $dh_menu "${list[@]}"; then
-            wlan_network="$result"
-        else
-            if wlan_error $"Error listing wireless networks"; then
-                continue
-            else
-                break
-            fi
-        fi
-
-        if d --menu $"Select authentication mode" 0 0 $dh_menu $"WPA-PSK" '' $"WPA-EAP" '' $"Open" ''; then
-            wlan_auth_mode="$result"
-        fi
-        wlan_auth_mode_conf=
-        if [ "$wlan_auth_mode" = "WPA-EAP" ]; then
-            wlan_auth_mode_conf=eap
-            if d --inputbox $"Username" 0 0; then
-                wlan_username="$result"
-            fi
-        fi
-        if [ "$wlan_auth_mode" = "WPA-PSK" ]; then
-                wlan_auth_mode_conf=psk
-        fi
-        if [ "$wlan_auth_mode" = "Open" ]; then
-                wlan_auth_mode_conf=open
-        fi
-        if [ "$wlan_auth_mode" != "Open" ]; then	
-            wlan_password=
-            if d --insecure --passwordbox  $"Password" 0 0; then
-                wlan_password="$result"
-            fi
-        fi
-        break
-    done
- 
-    config_file=`mktemp -qt 'firstboot-XXXXXX'`
-
-    cat << EOF > $config_file
-BOOTPROTO='dhcp'
-STARTMODE='auto'
-WIRELESS_AP_SCANMODE='1'
-WIRELESS_AUTH_MODE='$wlan_auth_mode_conf'
-WIRELESS_ESSID='$wlan_network'
-WIRELESS_MODE='Managed'
-EOF
-
-    if [ $wlan_auth_mode = "WPA-PSK" ]; then
-        echo "WIRELESS_WPA_PSK='$wlan_password'" >> $config_file
-    fi
-
-    if [ $wlan_auth_mode = "WPA-EAP" ]; then
-        echo "WIRELESS_WPA_IDENTITY='$wlan_username'" >> $config_file
-        echo "WIRELESS_WPA_PASSWORD='$wlan_password'" >> $config_file
-        echo "WIRELESS_EAP_AUTH='mschapv2'" >> $config_file
-    fi
-
-    run mv -f $config_file /etc/sysconfig/network/ifcfg-$wlan_device
-	
-fi
+call_module_hook systemd_firstboot
+call_module_hook post
+call_module_hook cleanup
```

Comment:
cleanup needs to run on exit, use `trap` for that.

Reply:
Haven't used trap for while but now it should be updated